### PR TITLE
Small documentation tweaks

### PR DESCRIPTION
**Pull request by tomverran**
Just made some code look more like PHP & briefly mentioned annotation caching

I'm afraid this doesn't update the Japanese README much, I don't speak it :(

### CODE REVIEW

**Comment by craigjbass** [comment]
:+1: Tom, exciting to have yet another clearbooksican involved.

**Comment by tomverran** [comment]
Hopefully I'll be able to make some more substantial contributions once I've understood how this all works!

**Comment by koriym** [comment]
Hello @tomverran , Thank you very much. exciting to have a contribution.

@tomverran @craigjbass Let me know where if you guys think we need code refactoring. I would like to support as much as I can. No problem for Japanese ;) I'll do it. But beware for my bad English, Thanks !

**Comment by craigjbass** [comment]
I think the only things that I would like to refactor are the usages of references - in my experience it's harder to debug/improve code that uses them as the logic is hard to follow. 

PHP uses copy-on-write memory handling, so we can take advantage of this instead, eliminating any reason to use reference for performance reasons.

**Comment by koriym** [comment]
No performance reason exists.
I will try it. Thanks.

**Comment by koriym** [comment]
@craigjbass I did. >> https://github.com/koriym/Ray.Di/commit/4bb3b9c2b2ad0ba0480ba0bf702b004b2ffdd96e
Hope this change helps you to make it easy more to read / modify.
